Core/Src/main.c

```diff
@@ -42,13 +42,19 @@
 /* USER CODE END PM */
 
 /* Private variables ---------------------------------------------------------*/
+SPI_HandleTypeDef hspi;
+TIM_HandleTypeDef htim1;
```

Comment:
I don't think this can pass the code compilation since you didn't include the corresponding header files to these data types.